### PR TITLE
make the `.line` and `inlined_at` fields in `LineInfoNode` `Int32`

### PR DESCRIPTION
**Pull request by KristofferC**
This reduces the size of `LineInfoNode` from 40 bytes to 32. Using `Int32` for line numbers is also already done in various other places (as can be seen by the diff in the PR) so this makes things more consistent as well.

### CODE REVIEW

File: stdlib/Serialization/src/Serialization.jl [hunk 1]
```diff
@@ -1112,7 +1112,8 @@ function deserialize(s::AbstractSerializer, ::Type{Core.LineInfoNode})
         method = mod
         mod = Main
     end
-    return Core.LineInfoNode(mod, method, deserialize(s)::Symbol, deserialize(s)::Int, deserialize(s)::Int)
+    T = format_version(s) >= 19 ? Int32 : Int
+    return Core.LineInfoNode(mod, method, deserialize(s)::Symbol, Int32(deserialize(s)::T), Int32(deserialize(s)::T))
```
Comment (JeffBezanson) [inline stdlib/Serialization/src/Serialization.jl:1116]: I would just allow either type here, so deserializing works even without a header specifying the version.

Comment (KristofferC) [reply stdlib/Serialization/src/Serialization.jl:1116]: Makes sense.

Comment (KristofferC) [reply stdlib/Serialization/src/Serialization.jl]: Is https://github.com/JuliaLang/julia/pull/44548/commits/bd6615ea85b5cdd60647446e88ff9261b2f512fe what you meant?

Comment (JeffBezanson) [reply stdlib/Serialization/src/Serialization.jl]: We can add a `::Union{Int32, Int}` typeassert; the compiler will split that very nicely.

**Pull request merged**